neqo-transport/tests/sim/drop.rs

```diff
@@ -0,0 +1,70 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(clippy::module_name_repetitions)]
+
+use super::{Node, Rng};
+use neqo_common::Datagram;
+use neqo_transport::Output;
+use std::fmt::{self, Debug};
+use std::time::Instant;
+
+/// A random dropper.
+pub struct Drop {
+    threshold: u64,
+    max: u64,
+    rng: Option<Rng>,
+}
+
+impl Drop {
+    /// Make a new random drop generator.  Each `drop` is called, this generates a
+    /// random value between 0 and `max` (exclusive).  If this value is less than
+    /// `threshold` a value of `true` is returned.
+    pub fn new(threshold: u64, max: u64) -> Self {
```

Comment:
nit: do this need to be public?

Reply:
It's not used right now, but the `percentage` function isn't really that good, as it doesn't allow for fractional percentages (for instance).